FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.DataClient/Api/FhirApiDataClient.cs

```diff
@@ -214,13 +214,13 @@ private async Task<string> GetResponseFromHttpRequestAsync(Uri uri, string acces
                         _logger.LogInformation(hrEx, "Failed to search from FHIR server: FHIR server {0} is not found.", _dataSource.FhirServerUrl);
                         break;
                     default:
-                        _diagnosticLogger.LogError(string.Format("Failed to search from FHIR server: Status code: {0}.", hrEx.StatusCode));
-                        _logger.LogInformation(hrEx, "Failed to search from FHIR server: Status code: {0}.", hrEx.StatusCode);
+                        _diagnosticLogger.LogError(string.Format("Failed to search from FHIR server: Status code: {0}. Reason: {1}", hrEx.StatusCode, hrEx.Message));
+                        _logger.LogInformation(hrEx, "Failed to search from FHIR server: Status code: {0}. Reason: {1}", hrEx.StatusCode, hrEx.Message);
                         break;
                 }
 
                 throw new FhirSearchException(
-                    string.Format(Resource.FhirSearchFailed, uri),
+                    string.Format(Resource.FhirSearchFailed, uri) + $"Reason:{hrEx.Message}",
```

Comment:
Just curious, do we need add the exception message to FhirSearchException message as the hrEx is already passed here.

Reply:
Other components (e.g., health checker) will catch this exception and log exception's message which will ignore inner exception.